CMakeLists.txt

```diff
@@ -1,43 +1,27 @@
+# Set the minimum version of CMake that's required 
 cmake_minimum_required(VERSION 3.12)
 
-project(maier-saupe-lc-hydrodynamics VERSION 0.1 LANGUAGES C CXX CUDA)
+project(maier-saupe-lc-hydrodynamics VERSION 0.1 LANGUAGES C CXX)
 
-# Read in external packages
-find_package(Boost 1.76 COMPONENTS program_options serialization REQUIRED)
-find_package(HighFive REQUIRED)
-find_package(deal.II 9.3.0 REQUIRED)
-deal_ii_initialize_cached_variables()
+# find Deal II
+find_package(deal.II 9.3.0
+  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
+  REQUIRED)
+# massage Deal II into a more modern CMake Target
+if(NOT TARGET deal.II::Interface)
+  add_library(deal.II::Interface INTERFACE IMPORTED GLOBAL)
+  set_target_properties(deal.II::Interface PROPERTIES
+    INTERFACE_INCLUDE_DIRECTORIES "${DEAL_II_INCLUDE_DIRS}"
+    INTERFACE_LINK_LIBRARIES "${DEAL_II_LIBRARIES_RELEASE}"
+    )
+endif()
 
-# Set directory variables
-set(LOCAL_INCLUDE_DIR /usr/local/include)
-set(INCLUDE_DIR ${PROJECT_SOURCE_DIR}/include)
-set(SOURCE_DIR ${PROJECT_SOURCE_DIR}/src)
-set(EXTERN_SOURCE_DIR ${PROJECT_SOURCE_DIR}/extern-src)
```

Comment:
Generally it is bad practice to define your own variables because CMake returns the empty string for undefined variables. This means it is very difficult to debug a typo in your CMake files.